File: Strings/Longest Palindromic Substring.cpp
class Solution
{
public:
	string longestPalindrome(string s)
	{
		if (s.size() <= 1)
		{
			return s;
		}

		int cnt = 0;
		int len = 0;
		int palstart = 0;
		int n = s.size();

		for (int k = 0; k < n; k++)
		{
			// odd len
			cnt = 1;
			int tempstart = k;
			int l = k - 1;
			int r = k + 1;
			while (l >= 0 and r < n and s[l] == s[r])
			{
				cnt += 2;
				tempstart = l;
				l--;
				r++;
			}
			if (cnt > len)
			{
				len = cnt;
				palstart = tempstart;
			}

			// odd len
			if (k > 0 and s[k - 1] == s[k])
			{
				tempstart = k - 1;
				int l = k - 2;
				int r = k + 1;
				cnt = 2;
				while (l >= 0 and r < n and s[l] == s[r])
				{
					cnt += 2;
					tempstart = l;
					l--;
					r++;
				}
				if (cnt > len)
				{
					len = cnt;
					palstart = tempstart;
				}
			}
		}

		return s.substr(palstart, len);
	}
};

File: JAVA Codes/Codes/Strings/Longest_Palindromic_Substring.java
package Codes.Strings;

public class Longest_Palindromic_Substring {
	
}

class Solution {
    public String longestPalindrome(String s) {
        
        int startans = 0;
        int charcnt = 0;
        int n = s.length();

        for(int k=0; k<n; k++){
            // odd len pal
            int cnt = 1;
            int tempstart = k;        
            int i = k-1; 
            int j = k+1;
            while(i >= 0 && j < n && s.charAt(i) == s.charAt(j)){
                cnt += 2;
                tempstart = i;
                i--;
                j++;
            }
            if(cnt > charcnt){
                charcnt = cnt;
                startans = tempstart;
            }

            // even len pal
            if(k > 0 && k < n && s.charAt(k-1) == s.charAt(k)){
                cnt = 2;
                tempstart = k-1;        
                i = k-2; 
                j = k+1;
                while(i >= 0 && j < n && s.charAt(i) == s.charAt(j)){
                    cnt += 2;
                    tempstart = i;
                    i--;
                    j++;
                }
                if(cnt > charcnt){
                    charcnt = cnt;
                    startans = tempstart;
                }
            }
        }

        return s.substring(startans, startans+charcnt);

    }
}